src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -108,15 +108,15 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
                 }
                 if (entry == null)
                 {
-                    requestContext.Logger.Info($"Skipping Instance discovery for {authorityInfo.AuthorityType} authority because is not enabled.");
+                    requestContext.Logger.Info(() => $"Skipping Instance discovery for {authorityInfo.AuthorityType} authority because is not enabled.");
                     entry = CreateEntryForSingleAuthority(authorityInfo.CanonicalAuthority);
                 }
 
                 return entry;
             }
             else
             {
-                requestContext.Logger.Info($"Skipping Instance discovery for {authorityInfo.AuthorityType} authority because is not supported.");
+                requestContext.Logger.Info(() => $"Skipping Instance discovery for {authorityInfo.AuthorityType} authority because is not supported.");
```

Comment:
```suggestion
                requestContext.Logger.Info(() => $"Skipping Instance discovery for {authorityInfo.AuthorityType} authority because it is not supported.");
```